action/sealedenvelope.go

```diff
@@ -7,19 +7,50 @@ import (
 	"github.com/iotexproject/iotex-proto/golang/iotextypes"
 	"github.com/pkg/errors"
 
+	"github.com/iotexproject/iotex-core/config"
 	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
 )
 
 // SealedEnvelope is a signed action envelope.
 type SealedEnvelope struct {
 	Envelope
+	encoding      iotextypes.Encoding
+	externChainID uint32
+	srcPubkey     crypto.PublicKey
+	signature     []byte
+}
 
-	srcPubkey crypto.PublicKey
-	signature []byte
+// rawHash returns the raw hash of embedded Envelope (this is the hash to be signed)
+// an all-0 return value means the transaction is invalid
+func (sealed *SealedEnvelope) rawHash() hash.Hash256 {
+	if IsRLP(sealed.encoding) {
+		tx, err := actionToRLP(sealed.Action())
+		if err != nil {
+			return hash.ZeroHash256
+		}
+		h, err := rlpRawHash(tx, sealed.externChainID)
```

Comment:
this is the only place we need to call config.ExternChainID()

Reply:
also rlpSignedHash()

Reply:
by saving externChainID in LoadProto() once, all later func calls like `rawHash(), Hash()` become self-contained so they don't need to call outside `config.ExternChainID()` again